lib/router/router.js

```diff
@@ -260,7 +260,12 @@ export default class Router extends EventEmitter {
     const jsonPageRes = await this.fetchRoute(route)
     const jsonData = await jsonPageRes.json()
 
-    if (jsonData.buildIdMismatch) {
+    const { buildId } = window.__NEXT_DATA__
+
+    if (
+      process.env.NODE_ENV === 'production' &&
+      buildId !== jsonPageRes.headers.get('Next-Build-Id')
```

Comment:
Is this something we can put in the response itself? I'd like to not rely on server response implementation details, paving the way for supporting 100% static builds for 3.0